File: 011_rgb_numbers/rgb.py
#!/usr/bin/python

import sys

converted = []
for i, line in enumerate(sys.stdin):
  colors = line.rstrip().split(',')

  for color in colors:
    digits = [int(digit) for digit in color.split('-')]
    
    is_valid = 1
    for digit in digits:
      if digit > 255:
        is_valid = 0

    if is_valid:
      result = '#'
      result += ''.join(["%0.2X" % digit for digit in digits])
    else:
      result = 'INVALID'
    converted.append(result)

# print result
print ','.join(converted) 

File: 011_rgb_numbers/rgb.rb
#!/usr/bin/env ruby

converted = []
ARGF.each_with_index do |line, idx|

    colors = line.split(',')

    colors.each do |color|
        digits = color.split('-')

        is_valid = 1        
        convert = []
        digits.each do |digit|
            digit = digit.to_i
            if digit > 255
                is_valid = 0
            end 
            convert.push digit
        end

        result = ''
        if is_valid > 0
            convert = convert.collect {|x| "%02X" % x }     
            result = "#" + convert.join('')
        else 
            result ='INVALID'
        end

        converted.push result
   end
end

puts converted.join(',')
